src/Analyzers/VisualBasic/Analyzers/IncludeParenthesesForMethodInvocations/VisualBasicIncludeParenthesesForMethodInvocationsDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,64 @@
+Imports Microsoft.CodeAnalysis.RemoveUnnecessaryParentheses
+Imports Microsoft.CodeAnalysis.Precedence
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.CodeStyle
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.VisualBasic.CodeStyle
+Imports Microsoft.CodeAnalysis.Operations
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.IncludeParenthesesForMethodInvocations
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend NotInheritable Class VisualBasicIncludeParenthesesForMethodInvocationsDiagnosticAnalyzer
+        Inherits AbstractBuiltInCodeStyleDiagnosticAnalyzer
+
+        Public Sub New()
+            MyBase.New(
+                diagnosticId:=IDEDiagnosticIds.IncludeParenthesesForMethodInvocationsDiagnosticId,
+                [option]:=VisualBasicCodeStyleOptions.IncludeParenthesesForMethodInvocations,
+                language:=LanguageNames.VisualBasic,
+                title:=New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Add_invocation_parentheses), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)))
+        End Sub
+
+        Protected Overrides Sub InitializeWorker(context As AnalysisContext)
+            context.RegisterOperationAction(AddressOf AnalyzeInvocations, OperationKind.Invocation)
+        End Sub
+
+        Private Sub AnalyzeInvocations(context As OperationAnalysisContext)
+            Dim includeParen As Boolean = context.Options.GetOption(VisualBasicCodeStyleOptions.IncludeParenthesesForMethodInvocations, context.Operation.Syntax.SyntaxTree, context.CancellationToken).Value
+            Dim node = context.Operation.Syntax
+            Dim operation = DirectCast(context.Operation, IInvocationOperation)
+            If TypeOf node IsNot InvocationExpressionSyntax Then
+                ' This should never happen I think?
+                ' Should it be removed or kept as defensive check?
+                Return
+            End If
+            Dim invocationExpression = DirectCast(node, InvocationExpressionSyntax)
+            If includeParen Then
+                If invocationExpression.ArgumentList Is Nothing Then
+                    ' Parentheses doesn't exist, ex: Dim x = FunctionCall
+                    '                                        ^^^^^^^^^^^^
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, node.GetLocation(), additionalLocations:={node.GetLocation()}))
+                End If
+            Else
+                If invocationExpression.ArgumentList?.Arguments.Any() Then ' Don't report if there are arguments, parentheses cannot be removed in that case.
+                    Return
+                End If
+                context.ReportDiagnostic(Diagnostic.Create(UnnecessaryWithSuggestionDescriptor, invocationExpression.ArgumentList.GetLocation(), additionalLocations:={node.GetLocation()}))
```

Comment:
we don't want to fade if the option is in 'silent' mode.

Reply:
The user should be able to change silent to suggestion with editorconfig right?
In that case, should I check if it's silent or not, then report `Descriptor` or `UnnecessaryWithSuggestionDescriptor` accordingly?

Reply:
i was wrong here too.  we should always fade for this case :)